### ISSUE TITLE
Clean up in soma-base modules

### ISSUE THREAD

**Issue by denisri** [brainvisa/testbidon#163]
---

Author Name: **Geffroy, Dominique** (Geffroy, Dominique)
Original Redmine Issue: 5547, https://bioproj.extra.cea.fr/redmine/issues/5547
Original Date: 2012-07-11

---

The module soma.wip.application contains the classes that are really used in axon whereas soma.application seems to contain more experimental code using traits and it is not used for the moment. 
It is a bit confusing. Maybe the name of the modules should be inverted. Moreover, the tests with traits should not be integrated in the next release because it is not finished. Maybe they have been moved from soma-traits projects too soon (soma-base is distributed while soma-traits is not).

Some modules may be out of date -> to be checked. 
Maybe soma-base and soma-qtgui could be joined in axon project ? Is it really useful to have a separated project for these modules ?

**Comment by denisri** [comment]
---

Original Redmine Comment
Author Name: **Riviere, Denis** (Riviere, Denis)
Original Date: 2012-07-11T14:49:43Z

---

bq. Maybe soma-base and soma-qtgui could be joined in axon project ? Is it really useful to have a separated project for these modules ?

soma-base and soma-qtgui are used in some features of pyaims and pyanatomist, which are independent from axon. So they need to be in separate projects.

Otherwise, I agree the .wip. modules need some cleanup.
And I am still not sure the 'wip modules' concept was a very good idea since things stay there for years, and probably for eternity.

**Comment by denisri** [comment]
---

Original Redmine Comment
Author Name: **Cointepas, Yann** (Cointepas, Yann)
Original Date: 2012-07-31T10:17:13Z

---

There  is no wip anymore in soma-base. There are some modules from Axon that are still in python/soma directory but we could change their location. During a soma reorganization, I moved from soma-base to axon some obsolete modules that are used only in Axon. I did not change the name of the modules at this stage. My first goal was to remove Signature from soma to replace it by traits. I had to keep wip.Application because it is used for Axon configuration (with Signature) and I had no time to change that. I hoped that soma.application and traits would be ready for this release but... We may remove some modules from soma-base 4.3 if we do not want to distribute them.
